server/repository/mongo/schedule_repo.py

```diff
@@ -0,0 +1,11 @@
+from domains.service_shift import ServiceShift
+from config.mongodb_config import get_db
+
+class ScheduleMongoRepo:
+    def __init__(self):
+        self.db = get_db()
+        self.collection = self.db.schedules
+
+    def list_by_shelter_id(self, shelter_id):
+        shifts = self.collection.find({"shelter_id": shelter_id})
+        return [ServiceShift.from_dict(s) for s in shifts]
```

Comment:
I wonder if ScheduleMongoRepo could extend ServiceShiftsMongoRepo. Essentially the functionality is the same and it will be working with the same data structure: ServiceShift. The only thing that you'll need to redefine is the constructor (which you are already doing). Then, in the /schedule API endpoint, you'd create a ScheduleMongoRepo instance and pass it to the service_shifts_list_use_case. You won't need to have a separate use case for the schedule, you can just reulse the service_shift_list_use_case.